### DISCUSSION TITLE
Use Prisma migrate dev without losing data

### DISCUSSION THREAD

**Discussion by nihiluis** [prisma/prisma#25025] [Q&A]
### Question

Hi, I lost my local migration files. If I use prisma db pull, I get my local schema file updated.

If I hit prisma migrate dev after, prisma wants to reset my database: _Drift detected: Your database schema is not in sync with your migration history._

I deleted the _prisma_migrations table, so it shouldnt find any migrations in the database. How can I avoid losing data here?

Edit: I misunderstood Prisma I think, I reset my DB and restored the data via a pg command.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS:  Windows
- Database:  Postgres
- Node.js version:  21.7.1

- Run `prisma -v` to see your Prisma version and paste it 5.17.0

**Comment by nurul3101** [comment]
Hi @nihiluis 👋 

To confirm, you are no longer facing this issue, right?

**Reply by nihiluis** [reply]
Yup, I closed as resolved.
